platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/DefaultFilePropertyFactory.java

```diff
@@ -193,6 +197,19 @@ public Provider<File> getAsFile() {
             return new MappingProvider<>(File.class, this, new ToFileTransformer());
         }
 
+        @Override
+        public void conventionFromAny(Provider<Object> file) {
+            convention(file.map(any -> {
```

Comment:
How about:

```suggestion
            convention(provider.map(value -> {
```

This would make it more understandable for me.

Reply:
My R$ 0,02 is that since `file` reflects the role of the parameter, and `provider` reflects the type, the former would be preferable (given that Java variables are explicitly typed). Maybe `fileProvider` would be a decent middle ground. I am fine with any of the 3 options, just sharing an opinion about parameter and local variable naming.

Reply:
Both comments seem like a good improvements, thanks!

I also renamed method to `conventionFromAnyFile` since we actually allow just some types of a file (either File or Directory) and not any object.

Reply:
I take that back, `file` is also reflecting kind of type information (though something that is not available to the language), and `conventionSource` or `source` would be a better name for conveying its role. But again, I am fine with any of the options, just promoting the general principle.

Reply:
I changed that parameter to `provider` so it's consisent with `Property.convention(Provider<? extends T> provider)`.

I think it's good enough :)